src/scales/scale.time.js

```diff
@@ -50,19 +50,36 @@ module.exports = function(Chart) {
 		return a - b;
 	}
 
-	function buildLookupTable(ticks, min, max, linear) {
-		var ilen = ticks.length;
+	/**
+	 * Returns an array of {time, pos} objects used to interpolate a specific `time` or position
+	 * (`pos`) on the scale, by searching entries before and after the requested value. Since the
+	 * value is linearly interpolated, only timestamps that break the time linearity are added,
+	 * meaning that in the best case, it contains only 2 objects: {min, 0} and {max, 1}. `pos` is
```

Comment:
Could you at the beginning of this method do something like the below? (not sure this is exactly right, but should give the idea)

    if (linear) {
      table.push({min, 0});
      table.push({max, 1});
      return table;
    }

Then you could simplify the section with the complicated logic a bit. Right now it's:

    if (Math.round((next + prev) / 2) !== curr) {
    	pos = linear ? (curr - min) / (max - min) : ilen > 1 ? i / (ilen - 1) : 0;
    	table.push({time: curr, pos: pos});
    }

I think it could just become:

    table.push({time: curr, pos: ilen > 1 ? i / (ilen - 1) : 0});

I think this has a few readability advantages. First, it would make it pretty explicit from reading the code what the lookup table looks like in the linear case. Right now it's not as immediately obvious from the code. Second, I think you can make the clarifying assumption that every point will break linearity in the series case, which would also simplify things by removing the `if` check. There may be some points that don't break linearity, but I'm assuming it doesn't hurt to add extra points to the lookup table.